server/utils/referrals/caseListUtils.ts

```diff
@@ -52,6 +61,57 @@ export default class CaseListUtils {
     ])
   }
 
+  static pagination(
+    path: Request['path'],
+    queryParamsExcludingPage: Array<QueryParam>,
+    pageNumber: Paginated<ReferralSummary>['pageNumber'],
+    totalPages: Paginated<ReferralSummary>['totalPages'],
+  ): GovukFrontendPaginationWithItems {
+    const oneIndexedPageNumber = pageNumber + 1
+
+    const pagination: GovukFrontendPaginationWithItems = { items: [] }
+
+    const numberPageLinkNumbers = Array.from(
```

Comment:
I'm finding this variable name a bit hard to parse. Is this getting the links for each page? I wonder if breaking this down into smaller variables might help improve readability here e.g. storing the `Set` value in a variable. Would `pageLinkNumbers` or even `pageNumbers` be enough here?

Reply:
`pageNumbers`